python/ray/serve/_private/router.py

```diff
@@ -410,6 +415,42 @@ async def shutdown(self):
         self._shutdown = True
 
 
+class SharedHandleMetricsPusher:
+    def __init__(self, controller_handle: ActorHandle):
+        self._controller_handler = controller_handle
+
+        self._metrics_pusher = MetricsPusher()
+        self._router_metrics_managers: weakref.WeakSet[
+            RouterMetricsManager
+        ] = weakref.WeakSet()
```

Comment:
add a comment explaining why this is a `WeakSet()`? Is there a risk of this getting GC`ed ?